src/views/Login.tsx

```diff
@@ -25,9 +25,13 @@ const Login = () => {
   const { trigger, isMutating, error } = useAuthenticateUser();
   const navigate = useNavigate();
 
-  const handleSubmit = async (
-    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
-  ) => {
+  useEffect(() => {
+    clearToken(dispatch);
+
+    console.log();
+  }, []);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     const response = await trigger({
```

Comment:
I think all the authentication process should be moved out of the `Login.tsx`. In the view there should be method like `loginWithCredentials(username, password)` that do all the job, so:
- make an request
- save the token
- maybe even redirect the user? But that can be also handled separately

Can we move it to the `AuthProvider`?

Reply:
yes, that would be more clean